### PR TITLE
Add native ES Modules build target

### PR DESCRIPTION
**Pull request by marvinhagemeister**
This PR adds a native ES Modules build output for consumers.

**todo:**

- [x] verify with e2e repo (commonjs)
- [x] verified with unannounced Preact project (native esm)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -45,8 +127,9 @@
     "test": "./run",
     "lint": "eslint . run",
     "types": "tsc",
-    "clean": "rimraf *.d.ts doc",
-    "prepublishOnly": "npm run clean && npm run types"
+    "build": "babel --out-dir dist/ --out-file-extension .mjs assert_utils.js browser_console.js browser_utils.js config.js curl_command.js email.js external_locking.js index.js loader.js locking.js main.js net_utils.js output.js promise_utils.js render.js results.js runner.js utils.js version.js",
```
Comment (jzaefferer) [inline package.json:130]: Maintaining this file list and the `exports` block above will be pretty important - maybe that should be mentioned in a CONTRIBUTING.md? Could be a start at least.

That should also include a note about using require in this repo, e.g. to never use inline sync require (as mentioned in the comment in version.js)

Comment (marvinhagemeister) [reply package.json:130]: I'm hoping to remove the file mess very soon by moving all files into an `src/` directory. Didn't want to do it with this PR as the diff would become too hard to follow.

I'll add a note about inline `require` calls 👍

File: babel-transform-commonjs-to-esm.js [hunk 2]
```diff
@@ -0,0 +1,204 @@
+function isRequireCall(t, node) {
```
[the remark this answers is not in the document]
Comment (marvinhagemeister) [reply babel-transform-commonjs-to-esm.js:1]: I've tried that plugin, but it throws a hard exception on any inline `require` calls. We need a single one in the `importFile` function. I need to do some more tests to be sure that we can remove it, that's why I wrote my own babel plugin. I'm hoping to sunset it sometime in the future once we (hopefully) move to TS as the ESM -> CJS conversion is much better supported in all tools than CJS -> ESM
